tests/Exception/Http/HttpTest.php

```diff
@@ -96,10 +97,100 @@ public static function dataGetClass() {
 				'code'     => '404',
 				'expected' => '\\' . Status404::class,
 			],
-			'integer 422: class does not exist' => [
-				'code'     => 422,
+			'integer 460: class does not exist' => [
+				'code'     => 460,
 				'expected' => $default,
 			],
 		];
 	}
+
+
+	/**
+	 * Test whether all HTTP exception class can be instantiated.
+	 *
+	 * @dataProvider dataInstantiateHttpException
+	 *
+	 * @param int $code HTTP status code.
+	 *
+	 * @return void
+	 */
+	public function testInstantiateHttpException($status_code, $expected_exception_class) {
+		$exception_class  = Http::get_class($status_code);
+		$exception_object = new $exception_class('testing-1-2-3');
+		$this->assertInstanceOf(Http::class, $exception_object);
+		$this->assertInstanceOf($expected_exception_class, $exception_object);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public static function dataInstantiateHttpException() {
```

Comment:
Ha, I was just looking at this when your new commit came in. I think we can possibly make this more robust by getting rid of some of the hard-coding of the cases ?

Also using a numeric sting for the array key will not work to make these named datasets as PHP will just interpret those as integers and  PHPUnit will then not recognize them as data set names. I have a feeling we were talking about something related earlier ? 😝 

What about something like this ?
```php
		$all_status_codes    = array_keys(HttpStatus::MAP);
		$non_exception_codes = [100, 101, 102, 103, 200, 201, 202, 203, 204, 205, 206, 207, 208, 226, 300, 301, 302, 303, 307, 308];

		$data = [];
		foreach ($all_status_codes as $status) {
			if (in_array($status, $non_exception_codes, true)) {
				$data['Status ' . $status] = [$status, StatusUnknown::class];
			} else {
				$data['Status ' . $status] = [$status, Http::class . '\Status' . $status];
			}
		}

		return $data;
```

(and yes, you already have the `StatusUnknown` class imported, so no need to prefix it with `Http\`).

Reply:
Follow up... I'm actually struggling with the validity of the test in principle as I cannot come up with a way to make the test fail....

```php
		// Invalid codes.
		$data['Status 600']            = [600, StatusUnknown::class];
		$data['Not a status - string'] = ['not a status', StatusUnknown::class];
		$data['Not a status - false']  = [false, StatusUnknown::class];
		$data['Not a status - float']  = [10.5, StatusUnknown::class];
		$data['Not a status - object'] = [new \stdClass(), StatusUnknown::class];
```

☝🏻 The above will all pass.

Reply:
The test was not meant to test input values, but rather to ensure every single class is instantiable. With exceptions, I've often seen the issue that fatal errors were only discovered in production for invalid exception classes because they were never triggered during testing. So here we're making sure that if such an exception gets triggered, it will not fatal because of a broken exception class.

Reply:
Discussed off-GitHub: that means these tests are not actually about the `WpOrg\Requests\Exception\Http` class (which this test class is testing), but about the individual status exception classes, which means that strictly speaking, this test should be moved to its own file and get a different `@covers` tag.